src/allotropy/parsers/revvity_kaleido/kaleido_structure.py

```diff
@@ -0,0 +1,369 @@
+            )
+        )
+
+
+@dataclass(frozen=True)
+class AnalysisResult:
+    analysis_parameter: str
+    results: dict[str, str]
+
+    @staticmethod
+    def create(reader: CsvReader) -> Optional[AnalysisResult]:
+        barcode_line = assert_not_none(
+            reader.drop_until_inclusive("^Barcode:(.+),"),
+            msg="Unable to find background information.",
+        )
+
+        analysis_parameter = None
+        for element in barcode_line.split(","):
+            if re.search("^.+:.+$", element):
+                key, value = element.split(":", maxsplit=1)
+                if "AnalysisParameter" in key:
+                    analysis_parameter = value
+                    break
+
+        analysis_parameter = assert_not_none(
+            analysis_parameter,
+            msg="Unable to find analysis parameter in Analysis Results section.",
+        )
+
+        try:
+            results = assert_not_none(
+                reader.pop_csv_block_as_df(header=0, index_col=0),
+                msg="Unable to find results table.",
+            )
+        except AllotropeConversionError:
+            logging.warning(
+                f"Unable to read analysis result '{analysis_parameter}'. Ignoring"
+            )
+            return None
+
+        for column in results:
+            if str(column).startswith("Unnamed"):
+                results = results.drop(columns=column)
+
+        try:
+            a1_str_value = str(results.iloc[0, 0])
+        except IndexError as e:
+            error = f"Unable to find first position of analysis result '{analysis_parameter}'."
+            raise AllotropeConversionError(error) from e
+
+        if try_float_or_none(a1_str_value) is None:
+            return None
+
+        return AnalysisResult(
+            analysis_parameter=analysis_parameter,
+            results={
+                f"{row}{col}": values[col]
+                for row, values in results.iterrows()
+                for col in results.columns
+            },
+        )
+
+    def get_image_feature_name(self) -> str:
+        return self.analysis_parameter
+
+    def get_result(self, well_position: str) -> str:
+        return str(
+            assert_not_none(
+                self.results.get(str(well_position)),
+                msg=f"Unable to get well at position '{well_position}' from analysis result '{self.analysis_parameter}'.",
+            )
+        )
+
+    def get_image_feature_result(self, well_position: str) -> float:
+        return try_float(
+            self.get_result(well_position),
+            f"analysis result '{self.analysis_parameter}' at '{well_position}'",
+        )
+
+
+@dataclass(frozen=True)
+class MeasurementInfo:
+    instrument_serial_number: str
+    measurement_time: str
+    protocol_signature: str
+    measurement_signature: str
+
+    @staticmethod
+    def get_instrument_serial_number(elements: dict[str, str]) -> str:
```

Comment:
You should instead follow the `create` pattern: make a single `create` method for Measurement info that gets all these values and returns a `MeasurementInfo`